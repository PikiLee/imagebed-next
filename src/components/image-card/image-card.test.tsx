import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { expect, it, vi } from 'vitest'

import ImageCard from './image-card'

it('should be able to copy url', async () => {
  const user = userEvent.setup()

  const onDelete = vi.fn()
  const imageUrl = 'http://localhost:3000/image.jpg'
  const screen = render(<ImageCard url={imageUrl} onDelete={onDelete} />)

  const spy = vi.spyOn(navigator.clipboard, 'writeText')
  const copyButton = screen.getByRole('button', { name: /copy url/i })
  await user.click(copyButton)
  expect(spy).toHaveBeenCalledWith(imageUrl)
  screen.unmount()
})

it('should be able to delete image', async () => {
  const user = userEvent.setup()

  const onDelete = vi.fn()
  const imageUrl = 'http://localhost:3000/image.jpg'
  const screen = render(<ImageCard url={imageUrl} onDelete={onDelete} />)

  const deleteButton = screen.getByRole('button', { name: /delete/i })
  await user.click(deleteButton)
  expect(onDelete).toHaveBeenCalledWith(imageUrl)

  screen.unmount()
})
