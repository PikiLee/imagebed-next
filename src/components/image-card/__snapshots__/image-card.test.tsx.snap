// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="rounded-lg border border-slate-200 bg-white text-slate-950 shadow-sm dark:border-slate-800 dark:bg-slate-950 dark:text-slate-50"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
    />
    <div
      class="p-6 pt-0"
    >
      <div
        class="relative"
      >
        <img
          alt="Uploaded Image"
          height="1080"
          src="http://localhost:3000/image.jpg"
          width="1920"
        />
        <div
          class="absolute top-4 right-4 flex gap-2"
        >
          <button
            aria-label="Copy URL"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-white transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-slate-950 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 dark:ring-offset-slate-950 dark:focus-visible:ring-slate-300 bg-slate-900 text-slate-50 hover:bg-slate-900/90 dark:bg-slate-50 dark:text-slate-900 dark:hover:bg-slate-50/90 h-10 px-4 py-2"
            data-state="closed"
          >
            <svg
              class="lucide lucide-copy "
              fill="none"
              height="16"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="14"
                rx="2"
                ry="2"
                width="14"
                x="8"
                y="8"
              />
              <path
                d="M4 16c-1.1 0-2-.9-2-2V4c0-1.1.9-2 2-2h10c1.1 0 2 .9 2 2"
              />
            </svg>
          </button>
          <button
            aria-label="Delete"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-white transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-slate-950 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 dark:ring-offset-slate-950 dark:focus-visible:ring-slate-300 bg-red-500 text-slate-50 hover:bg-red-500/90 dark:bg-red-900 dark:text-slate-50 dark:hover:bg-red-900/90 h-10 px-4 py-2"
            data-state="closed"
          >
            <svg
              class="lucide lucide-trash2 "
              fill="none"
              height="16"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 6h18"
              />
              <path
                d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
              />
              <path
                d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
              />
              <line
                x1="10"
                x2="10"
                y1="11"
                y2="17"
              />
              <line
                x1="14"
                x2="14"
                y1="11"
                y2="17"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="flex items-center p-6 pt-0"
    />
  </div>
</DocumentFragment>
`;
